### PR TITLE
Misc tests

### PR DESCRIPTION
**Pull request by sunfishcode**
A few more floating point tests, but also includes a test for memory size overflow.

### CODE REVIEW

File: ml-proto/test/memory.wast [hunk 1]
```diff
@@ -31,6 +32,10 @@
   (module (memory 1 2 (segment 0 "a") (segment 2 "b") (segment 1 "c")))
   "data segment not disjoint and ordered"
 )
+(assert_invalid
+  (module (memory 0 65536))
```
Comment (rossberg-chromium) [inline ml-proto/test/memory.wast]: How about also adding a test with a value that overflows the 32-bit (resp 64-bit range), to check that there is no accidental wrap-around?

**Comment by rossberg-chromium** [comment]
lgtm

File: ml-proto/test/memory.wast (hunk 1 above)
Comment (sunfishcode) [inline ml-proto/test/memory.wast]: Ok; tests added.
